convert/jenkinsjson/convert.go

```diff
@@ -973,6 +975,10 @@ func canMergeSteps(step1, step2 *harness.Step) bool {
 		return false
 	}
 
+	if !hasDefaultOrNoImage(exec1) || !hasDefaultOrNoImage(exec2) {
```

Comment:
The bullet points you laid out are what I expected, so there is a very good chance I am misunderstanding the boolean logic here